### PR TITLE
Add command to generate ansible hosts file

### PR DESCRIPTION
**Pull request by machacekondra**
This patch add support generate ansible hosts file, which reflect the hosts created by lago.

This will be usefull to integrate with Ansible instead of bash or python scripts.

### CODE REVIEW

**Comment by nvgoldin** [comment]
ci test please

File: lago/cmd.py [hunk 1]
```diff
@@ -398,6 +400,45 @@ def do_console(prefix, host, **kwargs):
...
+        ]
+        for ip in vm_mgmt_ip:
+            inventory[vm._spec['vm-type']].append(
+                "{ip} ansible_host={ip} ansible_ssh_private_key_file={ssh_key}\n".format(
+                    ip=ip,
+                    ssh_key=prefix.virt_env.prefix.paths.ssh_id_rsa()
+                )
+            )
+
+    with open('./ansible_hosts', 'w+') as f:
```
Comment (nvgoldin) [inline lago/cmd.py]: I think either allowing to pass a file descriptor as parameter or printing to stdout would be nice, allowing you to do something like:
```
lago ansible_hosts > my_file
```

Comment (machacekondra) [inline lago/cmd.py]: I agree, thanks.

**Comment by nvgoldin** [comment]
@machacekondra  - sorry, forgot to mention - also need pep8. Good combination is ``autopep8 && yapf ...``, I'm testing this now, if you can also rebase on master, as I'm going to start the release probably today, so we'll get it in for v0.32.

+1 for changing the print function, if you can put it on a different patch(in the same PR) - would be nice, though not critical.

**Comment by machacekondra** [comment]
This is an example output:

```bash
 $ sudo lagocli ansible_hosts
current session does not belong to lago group.
[ovirt-engine]
192.168.201.3 ansible_host=192.168.201.3 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
[ovirt-host]
192.168.201.2 ansible_host=192.168.201.2 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
192.168.201.4 ansible_host=192.168.201.4 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
```

File: lago/cmd.py [hunk 2]
```diff
@@ -398,6 +400,44 @@ def do_console(prefix, host, **kwargs):
...
+    1.2.3.4 ansible_host=1.2.3.4 ansible_ssh_private_key_file=/path/id_rsa
+    1.2.3.6 ansible_host=1.2.3.6 ansible_ssh_private_key_file=/path/id_rsa
+    [ovirt-engine]
+    1.2.3.5 ansible_host=1.2.3.5 ansible_ssh_private_key_file=/path/id_rsa
+    """
+    inventory = defaultdict(list)
+    for vm in prefix.virt_env.get_vms().values():
+        vm_mgmt_ip = [
+            nic.get('ip') for nic in vm.nics()
+            if 'management' in nic.get('net')
```
Comment (nvgoldin) [inline lago/cmd.py]: This prints nothing for me, I think the management properties are in the ``prefix.get_nets()``, and then you can use ``nets['net_name'].is_management()``.

File: lago/cmd.py [hunk 3]
```diff
@@ -398,6 +400,44 @@ def do_console(prefix, host, **kwargs):
...
+    hosts configuration. It defines an IP address and private key
+    for the machine in group named by its type, it is printed to
+    standard output. The content looks for example like this:
+
+    [ovirt-host]
+    1.2.3.4 ansible_host=1.2.3.4 ansible_ssh_private_key_file=/path/id_rsa
+    1.2.3.6 ansible_host=1.2.3.6 ansible_ssh_private_key_file=/path/id_rsa
+    [ovirt-engine]
+    1.2.3.5 ansible_host=1.2.3.5 ansible_ssh_private_key_file=/path/id_rsa
+    """
```
Comment (nvgoldin) [inline lago/cmd.py]: This doesn't format nicely in the CLI :/
```
positional arguments:
  VERB
    status
    copy-from-vm        Copy file from a virtual machine to local machine
    shell               Open shell on the domain or run as script/command
    console             Open serial console to the domain
    deploy
    copy-to-vm          Copy file/dir to a virtual machine from the local host
    revert              Revert resources to a snapshot
    ansible_hosts       This method iterate through all the VMs and create an
                        ansible hosts configuration. It defines an IP address
                        and private key for the machine in group named by its
                        type, it is printed to standard output. The content
                        looks for example like this: [ovirt-host]
                        192.168.201.2 ansible_host=192.168.201.2
                        ansible_ssh_private_key_file=/path/id_rsa
                        192.168.201.4 ansible_host=192.168.201.4
                        ansible_ssh_private_key_file=/path/id_rsa [ovirt-
                        engine] 192.168.201.3 ansible_host=192.168.201.3
                        ansible_ssh_private_key_file=/path/id_rsa
```
Also - I'm not sure about the VM types as keys, if not using ovirtlago, they would all be ``default``, maybe using the name then would be better then(if there is no specific type).

File: lago/cmd.py (hunk 2 above)
Comment (nvgoldin) [inline lago/cmd.py]: The same init file I used:
```yaml
domains:
  puppet-tester:
    vm-type: ovirt-engine
    cpu_model: Opteron_G1
    memory: 4096
    service_provider: systemd
    nics:
      - net: lago
    disks:
      - template_name: el7.3-base
        type: template
        name: root
        dev: vda
        format: qcow2
      - comment: mount tester /dev/sdb
        size: 10G
        type: empty
        name: mnt
        dev: sdb
        format: raw

nets:
  lago:
    type: nat
    dhcp:
      start: 100
      end: 254
    management: true
    dns_domain_name: lago.local
```

File: lago/cmd.py (hunk 3 above)
Comment (machacekondra) [inline lago/cmd.py]: Ah, I thought it's take from the help='XXXX', not from the docstring.

The reason I've choosen the type of the vm, is for tasks such as:

```yaml
- name: Install cockpit on hosts
  yum:
    state: absent
    name: cockpit-ovirt-dashboard
  delegate_to: ovirt-host
```

I need some common group to same VMs.

Comment (nvgoldin) [inline lago/cmd.py]: Ok, makes sense. We can improve it later, I'll open an issue.

**Comment by ifireball** [comment]
I wonder if this won't be better implemented as a 'lago status' output filter.

WRT Ansible support, I'd like to see something more comprehensive then just generating a hosts list. How does Vagrant support Ansible for example? Maybe some of this work can be superseded by issue #410.

Also see previous discussion on #281

**Comment by nvgoldin** [comment]
@ifireball  - I sure hope this will only be the beginning :) meaning to say, lets get something going, get some use-cases for it(ansible) and according to the output and feedback we can extend. 
I also opened - https://github.com/lago-project/lago/issues/429

**Comment by machacekondra** [comment]
Vagrant generates following file:
```bash
$ cat .vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory 
# Generated by Vagrant

engine ansible_ssh_host=192.168.121.220 ansible_ssh_port=22 ansible_ssh_user='vagrant' ansible_ssh_private_key_file='/home/omachace/workspace/tests/ovirt/.vagrant/machines/engine/libvirt/private_key'
storage ansible_ssh_host=192.168.121.85 ansible_ssh_port=22 ansible_ssh_user='vagrant' ansible_ssh_private_key_file='/home/omachace/workspace/tests/ovirt/.vagrant/machines/storage/libvirt/private_key'
host1 ansible_ssh_host=192.168.121.130 ansible_ssh_port=22 ansible_ssh_user='vagrant' ansible_ssh_private_key_file='/home/omachace/workspace/tests/ovirt/.vagrant/machines/host1/libvirt/private_key'
host2 ansible_ssh_host=192.168.121.53 ansible_ssh_port=22 ansible_ssh_user='vagrant' ansible_ssh_private_key_file='/home/omachace/workspace/tests/ovirt/.vagrant/machines/host2/libvirt/private_key'
```

This patch does basically the same.

**Comment by ifireball** [comment]
Doesn't Vagrant have something more comprehensive that runs Ansible for you? I think it can at least run Ansible for you when you do `vagrant deploy`

**Comment by machacekondra** [comment]
The correct way to implement this would be in my opinion to create something like "provisioners" (that's how vagrant do that).

You can have `bash` provisioner `ansible` provisioner `puppet`, etc.
So I guess lago should have something like:
```yaml
provisioners:
  ansible:
    - path: xyz.yml
      parameters: addititonal cmd line parameters
  bash:
    - path: script.sh
      parameters: addititonal cmd line parameters
```

Or anything similar.

**Comment by machacekondra** [comment]
Now it looks like:

```
$ lagocli ansible_hosts
[ovirt-engine]
lago-basic-suite-master-engine ansible_host=192.168.201.3 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
[ovirt-host]
lago-basic-suite-master-host0 ansible_host=192.168.201.2 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
lago-basic-suite-master-host1 ansible_host=192.168.201.4 ansible_ssh_private_key_file=/home/omachace/workspace/ovirt-system-tests/deployment-basic-suite-master/default/id_rsa
```

**Comment by machacekondra** [comment]
@mwperina

**Comment by mwperina** [comment]
Hi,
lack of those information blocks adding Ansible modules tests into ovirt-system-tests. So can we merge this simple patch to be able to finish that task and later when more mature infrastructure is finished we can adapt existing Ansible tests to it?

Thanks

Martin

File: lago/cmd.py [hunk 4]
```diff
@@ -398,6 +401,50 @@ def do_console(prefix, host, **kwargs):
...
+    # This method iterate through all the VMs and create an ansible
+    # host inventory. It defines an IP address and private key
+    # for the machine in group named by its type, it is printed to
+    # standard output. The content looks for example like this:
+    #
+    # [ovirt-host]
+    # lago-host1 ansible_host=1.2.3.4 ansible_ssh_private_key_file=/path/id_rsa
+    # lago-host2 ansible_host=1.2.3.6 ansible_ssh_private_key_file=/path/id_rsa
+    # [ovirt-engine]
+    # lago-engin ansible_host=1.2.3.5 ansible_ssh_private_key_file=/path/id_rsa
```
Comment (mykaul) [inline lago/cmd.py]: lago-engin -> lago-engine

File: lago/cmd.py [hunk 5]
```diff
@@ -398,6 +401,50 @@ def do_console(prefix, host, **kwargs):
...
+    help='Create Ansible host inventory of the environment'
+)
+@in_lago_prefix
+@with_logging
+def do_generate_ansible_hosts(prefix, out_format, **kwargs):
+    """
+    Create Ansible host inventory of the environment
+    """
+    #
+    # This method iterate through all the VMs and create an ansible
```
Comment (mykaul) [inline lago/cmd.py]: iterate -> iterates
create -> creates
ansible -> Ansible

**Comment by nvgoldin** [comment]
@machacekondra 👍  looks good!

A run example(maybe we can add something similar later to the docs):
LagoInitFile:
```yaml
domains:
  vm01-g1:
    vm-type: ovirt-engine
    cpu_model: Opteron_G1
    memory: 2048
    service_provider: systemd
    nics:
      - net: lago
    disks:
      - template_name: el7.3-base
        type: template
        name: root
        dev: vda
        format: qcow2

  vm02-g2:
    vm-type: ovirt-host
    cpu_model: Opteron_G2
    memory: 2048
    service_provider: systemd
    nics:
      - net: lago
      - net: external
    disks:
      - template_name: el7.3-base
        type: template
        name: root
        dev: vda
        format: qcow2

  vm03-sandybridge:
    cpu_model: SandyBridge
    memory: 2048
    service_provider: systemd
    nics:
      - net: lago
      - net: external
    disks:
      - template_name: el7.3-base
        type: template
        name: root
        dev: vda
        format: qcow2
nets:
  lago:
    type: nat
    dhcp:
      start: 100
      end: 254
    management: true
    dns_domain_name: lago.local
  external:
    type: nat
    dhcp:
      start: 100
      end: 254
    management: false
    dns_domain_name: lago.local
```

```shell
> lago init && lago start
...
> lago --ignore-warnings ansible_hosts > ansible_hosts
> cat ansible_hosts
[default]
vm03-sandybridge ansible_host=192.168.200.4 ansible_ssh_private_key_file=/home/ngoldin/src/nvgoldin.github.com/lago-init-files/.lago/default/id_rsa
[ovirt-engine]
vm01-g1 ansible_host=192.168.200.2 ansible_ssh_private_key_file=/home/ngoldin/src/nvgoldin.github.com/lago-init-files/.lago/default/id_rsa
[ovirt-host]
vm02-g2 ansible_host=192.168.200.3 ansible_ssh_private_key_file=/home/ngoldin/src/nvgoldin.github.com/lago-init-files/.lago/default/id_rsa
> ansible ovirt-engine -u root -i ansible_hosts -m shell -a 'lscpu | grep "Model name"'
vm01-g1 | SUCCESS | rc=0 >>
Model name:            AMD Opteron 240 (Gen 1 Class Opteron)
>ansible ovirt-host -u root -i ansible_hosts -m shell -a 'lscpu | grep "Model name"'
vm02-g2 | SUCCESS | rc=0 >>
Model name:            AMD Opteron 22xx (Gen 2 Class Opteron)
>ansible vm03-sandybridge -u root -i ansible_hosts -m shell -a 'lscpu | grep "Model name"'
vm03-sandybridge | SUCCESS | rc=0 >>
Model name:            Intel Xeon E312xx (Sandy Bridge)
```

**Comment by nvgoldin** [comment]
ci merge please

**Comment by machacekondra** [comment]
Great, thank you for merging!

**Pull request merged**